### PR TITLE
Fix parameter assignment in error handling function

### PR DESCRIPTION
**Pull request by mdazam1942**

### CODE REVIEW

File: stix_shifter_modules/crowdstrike/stix_transmission/connector.py [hunk 1]
```diff
@@ -31,45 +36,76 @@ def _handle_errors(self, response, return_obj):
...
-            raise Exception(return_obj)
+        elif response_code >= 400:
+            if response_type == 'application/json':
+                error_response = json.loads(response_txt)
+                response_dict['type'] = 'ValidationError'
+                response_dict['message'] = error_response['errors'][0]['message']
+                ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+                raise QueryException(return_obj)
+            elif response_type == 'text/html':
+                error = ConnectionError(f'Error connecting the datasource: {response_txt}')
```
Comment (delliott90) [inline stix_shifter_modules/crowdstrike/stix_transmission/connector.py:54]: Would `reponse_txt` contain the name of the connector (crowdstrike) or should it be added to the message?

File: stix_shifter_modules/crowdstrike/stix_transmission/connector.py [hunk 2]
```diff
@@ -31,45 +36,76 @@ def _handle_errors(self, response, return_obj):
...
+                    self.logger.error('Error connecting the datasource: ' + str(error_response))
+                    ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+                else:
+                    raise Exception(response_txt)
+            elif response_code == 400:
+                if response_type == 'application/json':
+                    error_response = json.loads(response_txt)
+                    response_dict['type'] = 'ValidationError'
+                    response_dict['message'] = error_response['errors'][0]['message']
+                    self.logger.error('Error connecting the datasource: ' + str(error_response))
```
Comment (delliott90) [inline stix_shifter_modules/crowdstrike/stix_transmission/connector.py:86]: Do we need to add the connector name here?

File: stix_shifter_modules/crowdstrike/stix_transmission/connector.py [hunk 3]
```diff
@@ -31,45 +36,76 @@ def _handle_errors(self, response, return_obj):
...
+                else:
+                    raise Exception(response_txt)
             else:
-                raise Exception(response_txt)
+                if response_type == 'application/json':
+                    response_error_ping = json.loads(response_txt)
+                    response_dict = response_error_ping['errors'][0]
+                    ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+                elif response_type == 'text/html':
+                    error = ConnectionError(f'Error connecting the datasource: {response_txt}')
```
Comment (delliott90) [inline stix_shifter_modules/crowdstrike/stix_transmission/connector.py:96]: Connector name?

File: stix_shifter_modules/crowdstrike/stix_transmission/connector.py [hunk 4]
```diff
@@ -31,45 +36,76 @@ def _handle_errors(self, response, return_obj):
...
+                    ErrorResponder.fill_error(return_obj, response_dict, error=error, connector=self.connector)
+                else:
+                    raise Exception(response_txt)
         except Exception as e:
             if response_txt is not None:
-                ErrorResponder.fill_error(return_obj, message='unexpected exception', connector=self.connector)
+                ErrorResponder.fill_error(return_obj, message='unexpected exception: ' + str(response_txt), connector=self.connector)
```
Comment (delliott90) [inline stix_shifter_modules/crowdstrike/stix_transmission/connector.py:102]: Missing connector name in message?

File: stix_shifter_modules/crowdstrike/stix_transmission/connector.py [hunk 5]
```diff
@@ -195,10 +230,13 @@ async def create_results_connection(self, query, offset, length):
...
+            return ex.args[0]
         except Exception as ex:
-            if response_txt is not None:
-                ErrorResponder.fill_error(return_obj, message='unexpected exception', connector=self.connector)
-                self.logger.error('can not parse response: ' + str(response_txt))
-            else:
-                raise ex
+            error_dict = {}
+            error_dict['type'] = 'AttributeError'
+            error_dict['message'] = 'Error while parsing API response: ' + str(ex)
```
Comment (delliott90) [inline stix_shifter_modules/crowdstrike/stix_transmission/connector.py:238]: connector name?

File: stix_shifter_modules/crowdstrike/stix_transmission/connector.py [hunk 6]
```diff
@@ -195,10 +230,13 @@ async def create_results_connection(self, query, offset, length):
...
+            ErrorResponder.fill_error(return_obj, error_dict, ['message'], connector=self.connector)
+            self.logger.error('unexpected exception: ' + str(ex))
```
Comment (delliott90) [inline stix_shifter_modules/crowdstrike/stix_transmission/connector.py:240]: name?

**Review by delliott90 — CHANGES_REQUESTED** [review]
Looks fine, I just want to make sure the connector name is appearing in each error that is returned.

File: stix_shifter_modules/crowdstrike/stix_transmission/connector.py (hunk 1 above)
Comment (mdazam1942) [reply stix_shifter_modules/crowdstrike/stix_transmission/connector.py:54]: it should be in the message when error message is constructed in the next function call.
`ErrorResponder.fill_error(return_obj, response_dict, error=error, connector=self.connector)`

Error message basically will look like this-
` "crowdstrike connector error => Connection error: Error connecting the datasource: <Returned errors from datasource>"`

File: stix_shifter_modules/crowdstrike/stix_transmission/connector.py (hunk 2 above)
Comment (mdazam1942) [reply stix_shifter_modules/crowdstrike/stix_transmission/connector.py:86]: yeah we can add connector name here in error logger.

File: stix_shifter_modules/crowdstrike/stix_transmission/connector.py (hunk 3 above)
Comment (mdazam1942) [reply stix_shifter_modules/crowdstrike/stix_transmission/connector.py:96]: ErrorResponder.fill_error() function ingest connector name for all error response

File: stix_shifter_modules/crowdstrike/stix_transmission/connector.py (hunk 4 above)
Comment (mdazam1942) [reply stix_shifter_modules/crowdstrike/stix_transmission/connector.py:102]: ErrorResponder.fill_error() function ingest connector name for all error response

File: stix_shifter_modules/crowdstrike/stix_transmission/connector.py (hunk 5 above)
Comment (mdazam1942) [reply stix_shifter_modules/crowdstrike/stix_transmission/connector.py:238]: ErrorResponder.fill_error() function ingest connector name for all error response

File: stix_shifter_modules/crowdstrike/stix_transmission/connector.py (hunk 6 above)
Comment (mdazam1942) [reply stix_shifter_modules/crowdstrike/stix_transmission/connector.py:240]: yeah will add connector name in self.logger.error()

**Review by delliott90 — APPROVED** [review]

**Pull request merged**